spec.bs

```diff
@@ -718,35 +780,81 @@ To <dfn>create a request</dfn> given a [=URL=] |url|, a [=string=] <dfn for="cre
+
+  1. If |responseBody| is null or failure, return false.
   1. Let |headers| be |response|'s [=response/header list=].
   1. If [=header list/getting a structured field value=] "X-Allow-Protected-Audience" from |headers| does not
-    return true, set |resource| to failure and return.
+    return true, return false.
   1. Let |headerMimeType| be the result of [=header list/extracting a MIME type=] from |headers|.
-  1. Set |resource| to failure and return, if any of the following conditions hold:
+  1. Return false if any of the following conditions hold:
     * |headerMimeType| is failure;
     * |mimeType| is "`text/javascript`" and |headerMimeType| is not a [=JavaScript MIME type=];
     * |mimeType| is "`application/json`" and |headerMimeType| is not a [=JSON MIME type=].
   1. Let |mimeTypeCharset| be |headerMimeType|'s [=MIME type/parameters=]["`charset`"].
-  1. Set |resource| to failure and return if any of the following conditions hold:
+  1. Return false if any of the following conditions hold:
     * If |mimeTypeCharset| does not [=map/exist=], or |mimeTypeCharset| is "utf-8", and
       |responseBody| is not [=UTF-8=] encoded;
     * If |mimeTypeCharset| is "us-ascii", and |responseBody| is not [=ascii string=].
-  1. Set |resource| to |responseBody|.
-1. Wait for |resource| to be set.
-1. Return |resource|.
+  1. Return true.
+</div>
 
+<div algorithm>
+To <dfn>fetch script</dfn> given a [=URL=] |url|:
+
+  1. Let |request| be the result of [=creating a request=] with |url|, "`text/javascript`", and
+    null.
+  1. Let |script| be null.
+  1. [=Fetch=] |request| with [=fetch/processResponseConsumeBody=] set to the following steps given
+    a [=response=] |response| and |responseBody|:
+    1. If [=validate fetching response=] with |response|, |responseBody| and "`text/javascript`"
+      returns false, set |script| to failure and return.
+    1. Set |script| to |responseBody|.
+  1. Wait for |script| to be set.
+  1. Return |script|.
+</div>
+
+<div algorithm>
+To <dfn>fetch trusted signals</dfn> given a [=URL=] |url|, and a [=boolean=] |isBiddingSignal|:
+
+  1. Let |request| be the result of [=creating a request=] with |url|, "`application/json`", and
+    null.
+  1. Let |signals| be null.
+  1. Let |dataVersion| be null.
+  1. Let |formatVersionString| be null.
+  1. [=Fetch=] |request| with [=fetch/processResponseConsumeBody=] set to the following steps given
+    a [=response=] |response| and |responseBody|:
+    1. If [=validate fetching response=] with |response|, |responseBody| and "`application/json`"
+      returns false, set |signals| to failure and return.
+    1. Let |headers| be |response|'s [=response/header list=].
+    1. Let |dataVersionString| be the result of [=header list/getting a structured field value=]
+      "Data-Version" from |headers|.
+    1. If |dataVersionString| is not null:
+      1. If |dataVersionString| [=string/starts with=] "`0`" but [=string/is=] not "`0`", set

```

Comment:
This is interesting - is there some way for the version string to be octal or whatever (e.g., "01")?

Reply:
I have no idea what this is.  The code looks like it simply parses as an integer:  https://source.chromium.org/chromium/chromium/src/+/refs/heads/main:content/services/auction_worklet/trusted_signals.cc;l=493;drc=9ba7e889bc57846a89add93a003e4443dfbeb736;bpv=1;bpt=1
I'm removing this line.

Reply:
I think this comes from this line [0] in the explainer:

"The version number should be formatted with only the digits [0-9] with no leading 0s and fit in a 32-bit unsigned integer."

Not sure how important that line is?

[0] https://github.com/WICG/turtledove/blob/main/FLEDGE.md#31-fetching-real-time-data-from-a-trusted-server:~:text=The%20version%20number%20should%20be%20formatted%20with%20only%20the%20digits%20%5B0%2D9%5D%20with%20no%20leading%200s%20and%20fit%20in%20a%2032%2Dbit%20unsigned%20integer.